backend/app/auth/routes.py

```
@@ -0,0 +1,195 @@
+        return AuthResponse(
+            access_token=access_token,
+            refresh_token=result["refresh_token"],
+            user=UserResponse(**result["user"])
+        )
+    except HTTPException:
+        raise
+    except Exception as e:
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail=f"Registration failed: {str(e)}"
+        )
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)